src/pages/DatasetCatalog.js

```diff
@@ -365,111 +364,189 @@ class DatasetCatalog extends Component {
                 thead({}, [
                   tr({}, [
                     th({}),
-                    th({ className: "table-titles dataset-color cell-size" }, ["Dataset Id"]),
-                    this.state.dataSetList.dictionary.map((dictionary, dIndex) => {
-                      return h(Fragment, { key: dIndex }, [
-                        th({ isRendered: dictionary.key !== 'Sample Collection ID', className: "table-titles dataset-color cell-size", id: dictionary.key }, [
-                          dictionary.key
-                        ])
-                      ]);
-                    }),
```

Comment:
@solideoglori ... this was the root of the complication. The dictionary is basically a list of columns, but it isn't the full list of columns, just some of them. So, if we want to rearrange some of the columns not defined in the dictionary, we'd have to do some complicated breaks to check the index, insert the undefined ones where we wanted them, and then continue on with looping over the rest of the dictionary elements. 

To get around this problem (and to solve a few other inefficiencies), I ditched the dictionary approach altogether. Instead, we just define our headers here as we want them. Then, when writing out the rows, we grab the property we want that matches the column. It should be trivial now to do any kind of re-arrangement/addition in the future.